### PR TITLE
Finish getUser + auto generate userTypeID

### PR DESCRIPTION
**Pull request by nguyenquocviet2007**
- Add method getUser
- Auto generate userTypeID
- Add route for getUser

### CODE REVIEW

File: server/controller/Users/UserQueries.js [hunk 1]
```diff
@@ -12,8 +12,8 @@ const mysql = require("mysql");
 module.exports = {
     register: (data, callBack) => {
         dbConfig.query(
-            `insert into users(username, password) 
-                        values(?,?)`,
+            `insert into users(username, password, userTypeID) 
+                        values(?,?,1)`,
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js]: auto generate which mean that's random number @nguyenquocviet2007  please fix this!

File: server/controller/Users/UserQueries.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ module.exports = {
         )
     },
     
-    getUserByUserName: (username, callback) => {
+    getUserByUserName: (username, callBack) => {
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js]: What is the purpose of this function? @nguyenquocviet2007

File: server/controller/Users/UserQueries.js [hunk 3]
```diff
@@ -42,5 +42,18 @@ module.exports = {
                 }
             }
         )
-    }
+    },
+
+    getUsers: callBack => {
+        dbConfig.query(
+          `SELECT userName, password from users`,
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js]:
```suggestion
         `SELECT * FROM users`
```

File: server/controller/Users/UserQueries.js (hunk 2 above)
Comment (nguyenquocviet2007) [reply server/controller/Users/UserQueries.js]: when users login, this function will check their username are available or not

File: server/controller/Users/UserQueries.js [hunk 4]
```diff
@@ -13,7 +15,7 @@ module.exports = {
     register: (data, callBack) => {
         dbConfig.query(
             `insert into users(username, password, userTypeID) 
-                        values(?,?,1)`,
+                        values(?,?,rand())`,
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js]: @nguyenquocviet2007 Please fix this, we only  use 0 and 1 for userTypeID, so we won't insert that randomly, just insert 0 from now for user. And please make your variable is a constant and reference to this query

**Pull request merged**